packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +920,1034 @@ moduleFor('Components test: curly components', class extends RenderingTest {
+
+    this.runTask(() => this.context.set('somecomponent', 'notsomecomponent'));
+
+    this.assertText('notsomecomponent');
+  }
+
+  ['@test non-block with properties on attrs']() {
+    this.registerComponent('non-block', {
+      template: 'In layout - someProp: {{attrs.someProp}}'
+    });
+
+    this.render('{{non-block someProp=prop}}', {
+      prop: 'something here'
+    });
+
+    this.assertText('In layout - someProp: something here');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout - someProp: something here');
+
+    this.runTask(() => this.context.set('prop', 'other thing there'));
+
+    this.assertText('In layout - someProp: other thing there');
+
+    this.runTask(() => this.context.set('prop', 'something here'));
+
+    this.assertText('In layout - someProp: something here');
+  }
+
+  ['@skip non block with properties overridden in init']() {
+    let instance;
+    this.registerComponent('non-block', {
+      ComponentClass: Component.extend({
+        init() {
+          this._super(...arguments);
+          instance = this;
+          this.someProp = 'value set in instance';
+        }
+      }),
+      template: 'In layout - someProp: {{someProp}}'
+    });
+
+    this.render('{{non-block someProp=prop}}', {
+      prop: 'something passed when invoked'
+    });
+
+    this.assertText('In layout - someProp: value set in instance');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout - someProp: value set in instance');
+
+    this.runTask(() => this.context.set('prop', 'updated something passed when invoked'));
+
+    this.assertText('In layout - someProp: updated something passed when invoked');
+
+    this.runTask(() => instance.set('someProp', 'update value set in instance'));
+
+    this.assertText('In layout - someProp: update value set in instance');
+
+    this.runTask(() => this.context.set('prop', 'something passed when invoked'));
+    this.runTask(() => instance.set('someProp', 'value set in instance'));
+
+    this.assertText('In layout - someProp: value set in instance');
+  }
+
+  ['@htmlbars rerendering component with attrs from parent'](assert) {
+    let willUpdate = 0;
+    let didReceiveAttrs = 0;
+
+    this.registerComponent('non-block', {
+      ComponentClass: Component.extend({
+        didReceiveAttrs() {
+          didReceiveAttrs++;
+        },
+
+        willUpdate() {
+          willUpdate++;
+        }
+      }),
+      template: 'In layout - someProp: {{someProp}}'
+    });
+
+    this.render('{{non-block someProp=someProp}}', {
+      someProp: 'wycats'
+    });
+
+    assert.equal(didReceiveAttrs, 1, 'The didReceiveAttrs hook fired');
+    this.assertText('In layout - someProp: wycats');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout - someProp: wycats');
+    assert.equal(didReceiveAttrs, 2, 'The didReceiveAttrs hook fired again');
+    assert.equal(willUpdate, 1, 'The willUpdate hook fired once');
```

Comment:
@chancancode added the N step. Just wanted to make sure the assertions are right.